test/e2e/inspect_test.go

```diff
@@ -125,7 +125,8 @@ spec:
 				"owner":           "cluster",
 				"reconcile_info":  "",
 				"reconcile_state": "ok",
-			})
+			}
+			expected = append(append([]map[string]string{expected[0]}, expectedEndpointSlice), expected[1])
```

Comment:
Can we make it more readable somehow?
I can think of 2 ways: 
One is to have different values for the Service, Endpoint and EndpointSlice and append them based on hasEndpointSlice.
Or we can use copy and do something like [this](https://play.golang.org/p/THJRkJtahsx).

Reply:
I actually did what you've done in the playground initially as well. I am not sure which one would be more suitable for legibility. I guess we can revisit this after the sorting issue is resolved as the order might change and this might be unnecessary.

Reply:
Basically, I was thinking that the test should be clear that we are expecting the array elements in certain order and that order should be clear. Even I am not satisfied with either of the approaches. But yeah, we can visit this later.

Reply:
Yes that makes complete sense